ibis/formats/pyarrow.py

```diff
@@ -353,8 +353,8 @@ def convert_table(cls, table: pa.Table, schema: Schema) -> pa.Table:
         desired_schema = PyArrowSchema.from_ibis(schema)
         pa_schema = table.schema
 
-        if pa_schema.names != schema.names:
-            table = table.rename_columns(schema.names)
+        if list(pa_schema.names) != (schema_names := list(schema.names)):
+            table = table.rename_columns(schema_names)
```

Comment:
Gonna put up a separate PR to delete this code path to make sure that's safe to do, since it doesn't seem to be hit in any tests.

Reply:
#9616